### ISSUE TITLE
[no-object-literal-type-assertion] <issue title>

### ISSUE THREAD

**Issue by bluelovers** [typescript-eslint/typescript-eslint#572]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-object-literal-type-assertionn": ["off"],
  }
}
```

```TS
_words2(words: IWordsAll[]): IWordsRuntime[]
	{
		let r = this._words_r1;

		return words.map(function (value, index, array)
		{
			// @ts-ignore
			if (value.no_regex)
			{
				return value;
			}

			if (Array.isArray(value) && (value.length == 2 || value.length >= 3))
			{
				value = {
					_source: value,

					s: value[0],
					r: value[1],

					flags: value[2] as string,
				} as IWordsOutput;
			}
```

> when add `as any then no error

![image](https://user-images.githubusercontent.com/167966/58529519-a838ef00-820d-11e9-9979-f3f307ac9993.png)

**Expected Result**

looks like this rule need a better message?
or maybe is a bug

**Actual Result**

![image](https://user-images.githubusercontent.com/167966/58529487-8b042080-820d-11e9-9548-9b38ce283eeb.png)

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `X.Y.Z` |
| `@typescript-eslint/parser`        | `X.Y.Z` |
| `TypeScript`                       | `X.Y.Z` |
| `ESLint`                           | `X.Y.Z` |
| `node`                             | `X.Y.Z` |
| `npm`                              | `X.Y.Z` |

**Comment by bradzacher** [comment]
Hi there!

A few things

1) you need to provide a valid title - `<issue title>` is not helpful in the slightest.
2) is that your actual config? If so - you spelled the rule name wrong, which is why your setting of `off` didn't work.
3) The rule is working as intended when turned on - please read [the rule documentation](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-object-literal-type-assertion.md)

```ts
value = {
  _source: value,
  s: value[0],
  r: value[1],
  lags: value[2] as string,
} as any as IWordsOutputz
         ^^^^^^^^^^^^^^^^
         This is a cast on an object literal, which triggers the rule
```
